src/event-engine/states/handshaking.ts

```diff
@@ -24,25 +36,50 @@ HandshakingState.on(subscriptionChange.type, (context, event) => {
   return HandshakingState.with({ channels: event.payload.channels, groups: event.payload.groups });
 });
 
-HandshakingState.on(handshakingSuccess.type, (context, event) =>
-  ReceivingState.with({
-    channels: context.channels,
-    groups: context.groups,
-    cursor: event.payload,
-  }),
+HandshakingState.on(handshakeSuccess.type, (context, event) =>
+  ReceivingState.with(
+    {
+      channels: context.channels,
+      groups: context.groups,
+      cursor: {
+        timetoken: context.cursor?.timetoken ?? event.payload.timetoken,
+        region: event.payload.region,
```

Comment:
Is it possible that `region` is known in the `context`? If it is, then it will be better to check conditionally.

Reply:
It's handshake success... So now we are using region provided by server and disregarding custom region. As per design description.